reconcile/test/utils/test_gitlab_api.py

```diff
@@ -468,3 +470,70 @@ def test_get_group_members(
     assert mocked_gitlab_api.get_group_members("group") == [
         {"user": "small", "access_level": None}
     ]
+
+
+def test_share_project_with_group_positive(
+    mocker: MockerFixture, mocked_gitlab_api: GitLabApi
+):
+    project = create_autospec(Project)
+    project.members = create_autospec(ProjectMemberManager)
+    project.members.all.return_value = [
+        create_autospec(ProjectMember, id=mocked_gitlab_api.user.id, access_level=40)
+    ]
+    mocker.patch(
+        "reconcile.utils.gitlab_api.GitLabApi.get_project"
+    ).return_value = project
```

Comment:
can set `mocked_gitlab_api.project = project`

Reply:
the value for `project` comes from `get_project` so needs to mock this . I think `mocked_gitlab_api.project = project` is valid in case we need `self.project`. In the methods we use `self.get_project()`.

Reply:
the naming is a little confusing, `mocked_gitlab_api` itself is not mock, it's the real `GitLabApi` object we want to test, we shouldn't patch another method belongs to `GitLabApi`, should setup `mocked_gl`